base/applications/mspaint/winproc.cpp

```diff
@@ -580,6 +582,120 @@ LRESULT CMainWindow::OnSysColorChange(UINT nMsg, WPARAM wParam, LPARAM lParam, B
     return 0;
 }
 
+typedef ULONG (WINAPI *FN_MAPISendMail)(LHANDLE, ULONG_PTR, lpMapiMessage, FLAGS, ULONG);
+typedef ULONG (WINAPI *FN_MAPISendMailW)(LHANDLE, ULONG_PTR, lpMapiMessageW, FLAGS, ULONG);
+
+ULONG CMainWindow::OpenMailerWithAttachmentW(LPCWSTR pszPathName)
+{
+    CStringW strFileTitle;
+    if (PathFileExistsW(pszPathName) && imageModel.IsImageSaved())
+    {
+        strFileTitle = PathFindFileNameW(pszPathName);
+    }
+    else // Not existing
+    {
+        // Delete the temporary file if any
+        if (g_szTempFile[0])
+        {
+            ::DeleteFileW(g_szTempFile);
+            g_szTempFile[0] = UNICODE_NULL;
+        }
+
+        // Get the name of a temporary file
+        WCHAR szTempDir[MAX_PATH];
+        ::GetTempPathW(_countof(szTempDir), szTempDir);
+        if (!::GetTempFileNameW(szTempDir, L"afx", 0, g_szTempFile))
+            return MAPI_E_FAILURE;
+
+        // Get the file type if already saved
+        const GUID *FileType = &GUID_NULL;
+        if (imageModel.IsImageSaved())
+        {
+            CImageDx img;
+            FileType = img.FileTypeFromExtension(PathFindExtensionW(g_szFileName));
+        }
+
+        // Save it to the temporary file
+        HBITMAP hbm = imageModel.CopyBitmap();
+        BOOL ret = ::SaveDIBToFile(hbm, g_szTempFile, FALSE, FileType);
+        ::DeleteObject(hbm);
+
+        if (!ret)
+        {
+            g_szTempFile[0] = UNICODE_NULL;
+            return MAPI_E_FAILURE; // Failure
+        }
+
+        // Use the temporary file 
+        pszPathName = g_szTempFile;
+
+        // Set the file title
+        if (imageModel.IsImageSaved())
+        {
+            strFileTitle = PathFindFileNameW(g_szFileName);
+        }
+        else
+        {
+            strFileTitle.LoadString(IDS_DEFAULTFILENAME);
```

Comment:
iirc when the file *exists* but may not be saved (again) yet, it's the actual name of the file that is used for the attachment. You can test yourself all that with standard Windows' MSPaint, by the way.

Reply:
Fixed.